frontend/src/lib/taxonomy.tsx

```diff
@@ -1896,3 +1896,13 @@ export function getFilterLabel(key: PropertyKey, type: TaxonomicFilterGroupType)
     const data = getCoreFilterDefinition(key, type)
     return (data ? data.label : key)?.trim() ?? '(empty string)'
 }
+
+export function getPropertyKey(value: string, type: TaxonomicFilterGroupType): string {
+    // Find the key by looking through the mapping
+    const group = CORE_FILTER_DEFINITIONS_BY_GROUP[type]
+    if (group) {
+        const foundKey = Object.entries(group).find(([_, def]) => (def as any).label === value || _ === value)?.[0]
+        return foundKey || value
+    }
+    return value
+}
```

Comment:
@pauldambra do you know any other way to use the labels/keys from taxonomy instead of duplicating everywhere? I feel like this should be done in a lot of places already so I wonder if this method is necessary.
Not a blocker since it is approved and it works :)

Reply:
keys are different than labels here. Key I'm treating as the `$device_type`, and label would be `Device Type`.

I didn't find a way in the codebase where this existed already, which is why I created this function 🤔 but if @pauldambra i'd be happy to fix it!